src/test/kotlin/uk/gov/justice/digital/hmpps/manageusersapi/resource/prison/UserControllerIntTest.kt

```diff
@@ -885,6 +885,186 @@ class UserControllerIntTest : IntegrationTestBase() {
+          ),
+        )
+        .exchange()
+        .expectStatus().isUnauthorized
+    }
+
+    @Test
+    fun `access forbidden when no role`() {
+      webTestClient.post().uri("/linkedprisonusers/general")
+        .headers(setAuthorisation(roles = listOf()))
+        .body(
+          fromValue(
+            mapOf(
+              "existingAdminUsername" to "TESTUSER1_ADM",
+              "generalUsername" to "TESTUSER1_GEN",
+              "defaultCaseloadId" to "BXI",
+            ),
+          ),
+        )
+        .exchange()
+        .expectStatus().isForbidden
+    }
+
+    @Test
+    fun `access forbidden when wrong role`() {
+      webTestClient.post().uri("/linkedprisonusers/general")
+        .headers(setAuthorisation(roles = listOf("ROLE_WRONG_ROLE")))
+        .body(
+          fromValue(
+            mapOf(
+              "existingAdminUsername" to "TESTUSER1_ADM",
+              "generalUsername" to "TESTUSER1_GEN",
+              "defaultCaseloadId" to "BXI",
+            ),
+          ),
+        )
+        .exchange()
+        .expectStatus().isForbidden
+    }
+
+    @Test
+    fun `access forbidden when wrong scope`() {
+      webTestClient.post().uri("/linkedprisonusers/general")
+        .headers(setAuthorisation(roles = listOf("ROLE_CREATE_ROLE"), scopes = listOf("read")))
+        .body(
+          fromValue(
+            mapOf(
+              "existingAdminUsername" to "TESTUSER1_ADM",
+              "generalUsername" to "TESTUSER1_GEN",
+              "defaultCaseloadId" to "BXI",
+            ),
+          ),
+        )
+        .exchange()
+        .expectStatus().isForbidden
+    }
+
+    @Test
+    fun `Link a General user to an existing Admin User`() {
+      val createLinkedGeneralRequest = CreateLinkedGeneralUserRequest("TESTUSER1_ADM", "TESTUSER1_GEN", "BXI")
+
+      nomisApiMockServer.stubCreateLinkedGeneralUser(createLinkedGeneralRequest)
+
+      val prisonStaffUser = webTestClient.post().uri("/linkedprisonusers/general")
+        .headers(setAuthorisation(roles = listOf("ROLE_CREATE_USER")))
+        .body(
+          fromValue(
+            mapOf(
+              "existingAdminUsername" to "TESTUSER1_ADM",
+              "generalUsername" to "TESTUSER1_GEN",
+              "defaultCaseloadId" to "BXI",
+            ),
+          ),
+        )
+        .exchange()
+        .expectStatus().isCreated
+        .expectBody(PrisonStaffUserDto::class.java)
+        .returnResult().responseBody!!
+
+      assertThat(prisonStaffUser.staffId).isEqualTo(100L)
+      assertThat(prisonStaffUser.firstName).isEqualTo("First")
+      assertThat(prisonStaffUser.lastName).isEqualTo("Last")
+      assertThat(prisonStaffUser.status).isEqualTo("ACTIVE")
+      assertThat(prisonStaffUser.primaryEmail).isEqualTo("f.l@justice.gov.uk")
+      assertThat(prisonStaffUser.generalAccount?.accountType).isEqualTo(PrisonUsageType.GENERAL)
+      assertThat(prisonStaffUser.adminAccount?.accountType).isEqualTo(PrisonUsageType.ADMIN)
```

Comment:
Need to assert the complete structure returned here i.e. all fields of general and admin account, including caseloads